azurerm/internal/services/cosmos/cosmosdb_account_resource_test.go

```diff
@@ -993,3 +1016,106 @@ func checkAccAzureRMCosmosDBAccount_basic(data acceptance.TestData, consistency
 		resource.TestCheckResourceAttrSet(data.ResourceName, "secondary_readonly_key"),
 	)
 }
+
+func checkAccAzureRMCosmosDBAccount_key_vault_uri(data acceptance.TestData, kind documentdb.DatabaseAccountKind, consistency documentdb.DefaultConsistencyLevel) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-cosmos-%d"
+  location = "%s"
+}
+
+data "azuread_service_principal" "cosmosdb" {
+  display_name = "Azure Cosmos DB"
+}
+
+data "azurerm_client_config" "current" {}
+
+resource "azurerm_key_vault" "test" {
+  name                = "acctestkeyvault%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  tenant_id           = data.azurerm_client_config.current.tenant_id
+  sku_name            = "standard"
```

Comment:
To get the CMK work, we will have to enable `soft_delete` and `purge_protection` (see the document [here](https://docs.microsoft.com/en-us/azure/cosmos-db/how-to-setup-cmk#configure-your-azure-key-vault-instance)). And these two features may requires a premium SKU (not very sure, you could have a try)

Reply:
Good catch, I did this in my manual test code already and it worked, so I'll update the test here for it